packages/core/theme/core-theme-browser-internal/src/core_context_provider.tsx

```diff
@@ -6,35 +6,31 @@
  * Side Public License, v 1.
  */
 
-import React, { FC } from 'react';
-import type { ThemeServiceStart } from '@kbn/core-theme-browser';
+import React, { FC, useMemo } from 'react';
 import type { I18nStart } from '@kbn/core-i18n-browser';
-import { CoreThemeProvider } from './core_theme_provider';
+import { composeProviders, KibanaThemeProvider } from '@kbn/react-kibana-context';
+import { ThemeServiceStart } from '@kbn/core-theme-browser/src/types';
 
 interface CoreContextProviderProps {
-  theme: ThemeServiceStart;
   i18n: I18nStart;
+  theme: ThemeServiceStart;
   globalStyles?: boolean;
 }
 
 /**
- * utility component exposing all the context providers required by core when integrating with react
+ * Utility component exposing all the context providers required by core when integrating with React.
  **/
 export const CoreContextProvider: FC<CoreContextProviderProps> = ({
   i18n,
-  theme,
   children,
+  theme,
   globalStyles = false,
 }) => {
-  // `globalStyles` and `utilityClasses` default values are inverted from that of `EuiProvider`.
-  // Default to `false` (does not add EUI global styles) because more instances use that value.
-  // A value of `true` (does add EUI global styles) will have `EuiProvider` use its default value.
-  const includeGlobalStyles = globalStyles === false ? false : undefined;
+  const Provider = useMemo(() => composeProviders([i18n.Context, KibanaThemeProvider]), [i18n]);
```

Comment:
`useMemo` does not guarantee that the value will be memoized:

<img width="759" alt="image" src="https://github.com/elastic/kibana/assets/82822460/7394f4c6-5aac-41ff-90dd-7e6510238912">

To really memoize I guess you would need to do some trickery with `useState` + `useEffect`.

Reply:
It was a TS lint suggestion, rather than an optimization.  `i18n` and other Providers don't change on its own, so I figured this was close to a `noop`.  I'll double-check this code, though.

Reply:
This was unnecessary, and removed.